src/app.js

```diff
@@ -43,29 +43,36 @@ app.get('/help', (req, res) => {
 })
 
 app.get('/weather', (req, res) => {
-    if (!req.query.address) {
+    if (!req.query.address && !req.query.lat && !req.query.lon) {
         return res.send({
-            error: 'No address provided!'
+            error: 'No location provided!'
         })
     }
-    geocode.geocode(req.query.address, (error, { latitude, longitude, location } = {}) => {
-        if (error) {
-            return res.send({ error })
-        }
-        forecast.forecast(latitude, longitude, (error, { description, temperature, feelslike, icon }) => {
+
+    if (req.query.address) {
```

Comment:
still weird. let's discuss in DM

Reply:
I changed the weather endpoint quite drastically, please check the current implementation.